frontend/pages/admin/IntegrationsPage/cards/MdmSettings/ScepPage/ScepPage.tsx

```diff
@@ -0,0 +1,379 @@
+                text="Controls > OS Settings > Custom settings"
+              />{" "}
+              to configure how SCEP certificates are delivered to your hosts.{" "}
+              <CustomLink
+                url="https://fleetdm.com/learn-more-about/setup-ndes"
+                text="Learn more"
+                newTab
+              />
+            </span>
+          </li>
+        </ol>
+      </div>
+    </>
+  );
+};
+
+interface IScepPageProps {
+  router: InjectedRouter;
+}
+
+interface INdesFormData {
+  scepUrl: string;
+  adminUrl: string;
+  username: string;
+  password: string;
+}
+
+interface INdesFormErrors {
+  scepUrl?: string | null;
+  adminUrl?: string | null;
+}
+
+export interface IFormField {
+  name: string;
+  value: string;
+}
+
+const ScepPage = ({ router }: IScepPageProps) => {
+  const { isPremiumTier, config, setConfig } = useContext(AppContext);
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const [formData, setFormData] = useState<INdesFormData>({
+    scepUrl: config?.integrations.ndes_scep_proxy?.url || "",
+    adminUrl: config?.integrations.ndes_scep_proxy?.admin_url || "",
+    username: config?.integrations.ndes_scep_proxy?.username || "",
+    password: config?.integrations.ndes_scep_proxy?.password || "",
+  });
+
+  const [formErrors, setFormErrors] = useState<INdesFormErrors>({});
+  const [saveButtonDisabled, setSaveButtonDisabled] = useState(true);
+  const [isUpdatingNdesScepProxy, setIsUpdatingNdesScepProxy] = useState(false);
+
+  const {
+    data: appConfig,
+    isLoading: isLoadingAppConfig,
+    refetch: refetchConfig,
+    isError: isErrorAppConfig,
+  } = useQuery<IConfig, Error, IConfig>(["config"], () => configAPI.loadAll(), {
+    select: (data: IConfig) => data,
+    onSuccess: (data) => {
+      setConfig(data);
+    },
+  });
+
+  useEffect(() => {
+    const areAllFieldsEmpty = Object.values(formData).every(
+      (val) => val === ""
+    );
+    const areAllFieldsComplete = Object.values(formData).every(
+      (val) => val !== ""
+    );
+
+    setSaveButtonDisabled(!areAllFieldsEmpty && !areAllFieldsComplete);
+  }, [formData]);
+
+  const onInputChange = ({ name, value }: IFormField) => {
+    setFormErrors((prev) => ({ ...prev, [name]: null }));
+    setFormData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const onFormSubmit = async (evt: React.MouseEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+
+    const scepUrlValid = validateUrl({ url: formData.scepUrl });
+    const adminUrlValid = validateUrl({ url: formData.adminUrl });
+    const newFormErrors = {
+      scepUrl:
+        scepUrlValid || formData.scepUrl === ""
+          ? undefined
+          : "Must be a valid URL.",
+      adminUrl:
+        adminUrlValid || formData.adminUrl === ""
+          ? undefined
+          : "Must be a valid URL.",
+    };
+
+    setFormErrors(newFormErrors);
+
+    const areAllFieldsEmpty = Object.values(formData).every(
```

Comment:
Can't you can just have this value define `isRemovingNdesScepProxy` below directly?

Reply:
done!